### PR TITLE
Fix and extend green mode

### PR DESCRIPTION
**Pull request by vxgmichel**
Related: issue #186

### CODE REVIEW

File: tango/asyncio_executor.py [hunk 1]
```diff
@@ -94,7 +90,7 @@ def submit(self, fn, *args, **kwargs):
 
     def execute(self, fn, *args, **kwargs):
         """Execute an operation and return the result."""
-        if self.loop._thread_id == get_ident():
+        if self.in_executor_context:
```
Comment (hardion) [inline tango/asyncio_executor.py]: Isn't self.in_executor_context a method returning a boolean?

**Comment by hardion** [comment]
Is there a way to install a testing version from this PR?

File: tango/asyncio_executor.py (hunk 1 above)
Comment (vxgmichel) [reply tango/asyncio_executor.py]: It's actually a property, but you're right, it doesn't really make the code any cleaner. I'll change that.

**Comment by vxgmichel** [comment]
> Is there a way to install a testing version from this PR?

Yes :)

```bash
$ conda create -n test python=3 boost gcc numpy -c tango-controls tango
$ source activate test
$ export BOOST_ROOT=$CONDA_PREFIX TANGO_ROOT=$CONDA_PREFIX ZMQ_ROOT=$CONDA_PREFIX OMNI_ROOT=$CONDA_PREFIX
$ python setup.py install
```

**Comment by vxgmichel** [comment]
@hardion Could you merge it once you've tested it? I'd like to release v9.2.3.

**Comment by hardion** [comment]
Thanks Vincent for the merge request. I had trouble to test this version caused by a different conda env. Maybe for the next PR.

**Comment by vxgmichel** [comment]
No problem, gevent 1.3 broke the tests for all the branches and I needed this PR to rebase them, that's why I had to merge it myself.

**Pull request merged**